### PR TITLE
Added route for POST /admin/newsletter issue #18

### PR DESCRIPTION
**Pull request by Zolo-Ryan**
Code completion by @nancy-byte

The pr includes a route for `sending mail to all the subscribed users` using `nodemailer` by the `admin`. However the middleware is not set.

### CODE REVIEW

File: server/src/router/newsletter.js [hunk 1]
```diff
@@ -1,10 +1,38 @@
...
+    const { subject, body } = req.body;
+
+    const SubscribedUsers = await User.find({ newsletterIsSubscribed: true });
+    if (!SubscribedUsers || SubscribedUsers.length == 0) {
+      return res.status(404).json({ message: "No subscribed user found" });
+    }
+
+    const recipients = SubscribedUsers.map((user) => user.email);
+
+    await transporter.sendMail({
```
Comment (Himasnhu-AT) [inline server/src/router/newsletter.js]: This method sends a single email with all user emails. which is a bad practice, as we are revealing our user email in public. Instead put them in bcc to hide emails

File: server/src/router/newsletter.js [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
\ No newline at end of file
+import express from "express";
+const router = express.Router();
+import transporter from "../config/nodemailer.js";
+import User from "../model/user.js";
+import { mailer } from "../controllers/newsletter.js";
+
+router.get("/", async (req, res) => {
+  res.send(await User.find({ newsletterIsSubscribed: true }));
+});
```
Comment (Himasnhu-AT) [inline server/src/router/newsletter.js:9]: Don't you think this API is needed?

thoughts?? @Zolo-Ryan

**Review by Himasnhu-AT — CHANGES_REQUESTED** [review]
Resolve the two reviews

File: server/src/router/newsletter.js (hunk 2 above)
Comment (Zolo-Ryan) [reply server/src/router/newsletter.js:9]: We didn't create that route. Just moved it down. It was from other prs

File: server/src/router/newsletter.js (hunk 1 above)
Comment (Zolo-Ryan) [reply server/src/router/newsletter.js]: Done

**Comment by Debatreya** [comment]
@Himasnhu-AT yeh krde ekbar

**Review by Himasnhu-AT — APPROVED** [review]

**Pull request merged**